app/models/submission.rb

```diff
@@ -134,4 +155,26 @@ def can_be_ineligible_for_evaluation
 
     errors.add(:judging_status, "must remain eligible for evaluation when evaluators are assigned")
   end
+
+  def calculate_evaluation_status
+    assigned_evaluators = evaluator_submission_assignments.where(status: :assigned)
+
+    submission_evaluations = evaluations.
+      joins(:evaluator_submission_assignment).
+      where(evaluator_submission_assignments: { submission_id: id, status: :assigned })
+
+    completed_count = submission_evaluations.where.not(completed_at: nil).count
+    in_progress_count = submission_evaluations.where(completed_at: nil).count
+    assigned_count = assigned_evaluators.count
+
+    return :not_started if assigned_count.zero?
+
+    if completed_count == assigned_count
+      :completed
+    elsif completed_count.positive? || in_progress_count.positive?
+      :in_progress
+    else
+      :not_started
+    end
```

Comment:
or I guess it could be collapsed further (less readable?)
```suggestion
    if assigned_count.zero? || (completed_count.zero? && in_progress_count.zero?)
      # nobody is assigned or no assignees have started or completed any evaluations
      :not_started
    elsif completed_count == assigned_count
      :completed
    else
      :in_progress
    end
```